exonum/src/events/network.rs

```diff
@@ -360,7 +381,8 @@ impl NetworkHandler {
                     })
                     .and_then(move |(socket, raw)| (Ok(socket), Self::parse_connect_msg(Some(raw))))
                     .and_then(move |(socket, message)| {
-                        if pool.contains(message.pub_key()) {
+                        let connection_limit = pool.count_outgoing() >= max_connections;
```

Comment:
These comparisons using `>=` here vs `<` in `can_create_connections()` slightly concern me. These different lines agree because `!(a < b)` ≡ `(a >= b)`, but someone some day may forget about it, change one line but not the other, and introduce an inconsistency.

I'd rewrite it to using `<` in both cases (e.g., `!connections_available`). Or even introduce a method on `ConnectionPool` but that'd be a bit overkill, I guess.

Reply:
I think I'll leave sign like this here since the same convention is used in the listener method where incoming connections are accepted. But I'll rename variable to clarify for future which sign should be used here.